### PR TITLE
Audio and layout fixes+improvements

### PR DESCRIPTION
**Pull request by artjomsR**
Originally I wanted to fix a single bug. In a typical software dev fashion, [one thing led to another](https://www.youtube.com/watch?v=AbSehcT19u0).. Originally I thought about creating several PRs but it quickly became apparent that testing them would be a nightmare so I've combined them on a single branch. We can test it thoroughly and merge it fully/partially. 

# Fix: Clear audio if searched word is not found
To reproduce:
Define a word that has audio. Then define another random string of letters that doesn't have a definition - the pronunciation remains.

It _is_ possible to define words/phrases even if they weren't found in a dictionary so I think it's better not to carry over the pronunciation.

# Feature: Play audio on double click
Oftentimes I would get only 1 audio for a word, so this is a simple addition to be able to hear it a few times. Just to decide if it's worth adding it to the card. Currently the only way to do it is to switch to another audio (doesn't apply if there's only a single audio file) or redefine the word - but then you lose changes made to definitions

# Fix discard_audio button making the app wider than it was previously
As of master, current app is slightly wider (at minimum width, in vertical mode) compared to 10.2. This fixes it by putting them in the same cell and aligning the button to the right.

# Feature: Show accent for audios where it is available
If Forvo accent is available, display it in brackets for every audio. See screenshots below. If no accent is present - pronunciation item is displayed the same way as previously.

I've moved Forvo code around a bit to store language and accent along with pronunciation for every item. This should also tidy up the code a little bit and make it more readable (e.g. there's only 1 regex now instead of 2) 

# Chore: Replace playsound with QMediaPlayer

# Feature: mp3/ogg DL option
As of now, it seems that you can replace mp3 <-> ogg in a Forvo link, it'll give you the same audio in a different format. Works both ways. The `audios/` URL part is still needed for some files but mp3 <-> ogg works regardless of the URL format.

image
image
image
image

image

### CODE REVIEW

File: vocabsieve/audio_player.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    Thread(target=lambda: self.player.play()).start()
+
+  def play_audio(self, name: str, data: Dict[str, str], lang: str) -> str:
+    audiopath: str = data.get(name, "")
+    if not audiopath:
+        return ""
+
+    if not audiopath.startswith("https://"):
+        content = QUrl(audiopath)
+        self.crossplatform_play_sound(content)
```
Comment (artjomsR) [inline vocabsieve/audio_player.py:25]: I wasn't sure how to test this if branch
 
I did test the same piece of code in a console and it worked for a local file

Comment (1over137) [reply vocabsieve/audio_player.py:25]: Note that this is meant to be used with local files
which can be tested by using the local audio library feature, such as https://lingualibre.org/datasets/ (download a zip and unpack and import it in the local resource manager).
Note that they are allowed to be in any format your system can support. At a minimum it should include mp3, ogg, wav, and opus if your system have the codecs for it.

Comment (1over137) [reply vocabsieve/audio_player.py:25]: I tested it, and this version does not work. What you want is:
`content = QUrl.fromLocalFile(audiopath)`

Comment (artjomsR) [reply vocabsieve/audio_player.py]: Updated

**Pull request merged**